lib/uploadcare/rails/configuration.rb

```diff
@@ -0,0 +1,40 @@
+require 'singleton'
+
+module Uploadcare
+  module Rails
+    class Configuration
+      include Singleton
+
+      CONFIG_GLOBAL_PARAMS = %w[
+        public_key locale live manual_start images_only preview_step crop image_shrink
+        clearable tabs input_accept_types preferred_types system_dialog multipart_min_size
+        locale locale_translations locale_pluralize secure_signature secure_expire preview_proxy
+        preview_url_callback cdn_base do_not_store audio_bits_per_second video_preferred_mime_types
+        video_bits_per_second camera_mirror_default
+      ]
+
+      attr_accessor *CONFIG_GLOBAL_PARAMS
+
+      def widget_parameters
+        CONFIG_GLOBAL_PARAMS.map do |param_name|
+          param_value = instance_variable_get("@#{param_name}")
+          next if param_value.nil?
+
+          param_value = handle_param_value(param_value)
+          "UPLOADCARE_#{param_name.upcase} = #{param_value};"
+        end.compact.join("\n")
+      end
```

Comment:
it seems, that it would be more convenient to have a separate namespace for widget settings and don't mix them with other project/app settings.
Also, there may be several Uploaders with separate settings.

Reply:
@dmitry-mukhin, what do you mean under `separate namespace` here?

Several Uploaders case can be solved by setting local variables for each `field_tag` (it will override global settings) or for each `uploaderObject` (in JS)

Reply:
what is this `Uploadcare.Rails.Configuration` class is about?

Reply:
This class is a "container" for application configuration params. It includes global UC widget settings and Rails application settings (like `cache_files`, `store/delete` files options). This class also parses widget params, so it can be placed in <head> tag on HTML-page.

Reply:
Yeah, this first three expressions makes sense:
```
conf.pub_key
conf.secret_key
conf.widget.image_shrink
```
But individual widgets are set in another places (like views, helpers).
End up with first three expressions